server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/cs/deployment/ServerExtensionDependenciesProcessor.java

```diff
@@ -0,0 +1,89 @@
+package org.jboss.as.clustering.infinispan.cs.deployment;
+
+import org.infinispan.persistence.spi.AdvancedCacheLoader;
+import org.infinispan.persistence.spi.AdvancedCacheWriter;
+import org.infinispan.persistence.spi.AdvancedLoadWriteStore;
+import org.infinispan.persistence.spi.CacheLoader;
+import org.infinispan.persistence.spi.CacheWriter;
+import org.infinispan.persistence.spi.ExternalStore;
+import org.jboss.as.server.deployment.Attachments;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.as.server.deployment.DeploymentUnitProcessingException;
+import org.jboss.as.server.deployment.DeploymentUnitProcessor;
+import org.jboss.as.server.deployment.ServicesAttachment;
+import org.jboss.as.server.deployment.module.ModuleDependency;
+import org.jboss.as.server.deployment.module.ModuleSpecification;
+import org.jboss.modules.Module;
+import org.jboss.modules.ModuleIdentifier;
+import org.jboss.modules.ModuleLoader;
+
+/**
+ * Adds dependency from deployed module to Infinispan core.
+ *
+ * @author Sebastian Laskawiec
+ * @since 7.2
+ */
+public class ServerExtensionDependenciesProcessor implements DeploymentUnitProcessor {
```

Comment:
Yes, but it is present in `endpoint` module whereas implementation is in `infinispan`.